spec/validators/exhibit_id_validator.rb

```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe ExhibitIdValidator do
+  subject { described_class.new }
+
+  describe "#validate" do
+    let(:errors) { double("Errors") }
+    before do
+      allow(subject).to receive(:existing_exhibit_ids).and_return(['slug1'])
```

Comment:
If that turns out to be too complicated, maybe a query object to encapsulate it?